### ISSUE TITLE
`/intercept-info` rest api returns `"intercepted": false` event though there is an active interception.

### ISSUE THREAD

**Issue by mleclercq** [telepresenceio/telepresence#3240]
**Describe the bug**
While I have a "all TCP requests" interception on a pod, if I query the `/intercept-info` on the traffic-agent, I get `{"intercepted":false,"clientSide":false}` as a response, while I would expect to have `"intercepted": true`.

I tested several variants: with `?containerPort=9000` (the intercepted container port), with `?containerPort=0` or without any query parameters. I always get the same result.

In the debug logs of the traffic agent I get the following:

For `curl localhost:31415/intercept-info?containerPort=9000`, I get:

```
2023-06-15 21:39:08.7253 debug   API-server : Received /intercept-info
2023-06-15 21:39:08.7254 debug   API-server : no match found for path "", port 9000, map[Accept:[*/*] User-Agent:[curl/7.64.0]]
```

For `curl localhost:31415/intercept-info?containerPort=0` or `curl localhost:31415/intercept-info`, I get:

```
2023-06-15 21:38:37.8518 debug   API-server : Received /intercept-info
2023-06-15 21:38:37.8521 debug   API-server : no match found for path "", map[Accept:[*/*] User-Agent:[curl/7.64.0]]
```

For `curl localhost:31415/intercept-info?containerPort=1234` (i.e. wrong port), I get:

```
2023-06-15 21:39:08.7253 debug   API-server : Received /intercept-info
```
No "no match found" log in that case

**To Reproduce**
1. Set `telepresenceAPI.port = 31415`
1. run an intercept on a workload
2. `kubectl exec` in the intercepted pod
3. run `curl localhost:31415/intercept-info`

**Expected behavior**
`intercept-info` should return `"intercepted": true` when there is an active interception.

**Versions (please complete the following information):**
 - Output of `telepresence version`:

```
telepresence version       
An update of telepresence from version 2.13.1 to 2.14.0 is available. Please visit https://www.getambassador.io/docs/telepresence/latest/install/upgrade/ for more info.
Client         : v2.13.1
Root Daemon    : v2.13.1
User Daemon    : v2.13.1
Traffic Manager: v2.13.1
```
 - Operating system of workstation running `telepresence` commands: MacOS
 - Kubernetes environment and Version [e.g. Minikube, bare metal, Google Kubernetes Engine]: GKE

**Additional context**

I searched in the code a bit and found that the  "no match found for path" log seems to come from this function:

https://github.com/telepresenceio/telepresence/blob/237f965af668675f0e66059a241ce02cd03f565a/cmd/traffic/cmd/agent/fwdstate.go#L39-L55

If `containerPort == 0` then the function should return at line 43 and not print the log.
Also on line 50 if `containerPort == 0` then the `"port %d"` segment is not added to the log. Indeed, when I test with `?containerPort=0` I do not have the "port " portion in the log and when I test with `?containerPord=9000` I get "port 9000" in the log. This means that when I test `?containerPort=0`, the log tells me that `containerPort == 0` but if that's true, then the function should have returned at line 43 and I shouldn't have a log 🤯

Something doesn't add up, so I suspect I'm not looking at the right code...

**Comment by thallgren** [comment]
Do you get the same behavior with a personal intercept, or is this specific to the "All TCP requests" kind of intercept?

**Comment by mleclercq** [comment]
With a personal intercept it seems to work but ATM this is not appropriate for my use-case since we don't have the machinery in place to forward the intercept headers to that particular workload which is pretty deep in our stack.

**Comment by mleclercq** [comment]
I managed to make the `/intercept-info` rest api to work but for that I had to deploy the traffic-manager with the following values:

```
image.name: tel2
agent.image.name: tel2
```

That is AFAIU using the OSS images. I tried that because, as mentioned above, I noticed that the debug logs I was getting didn't match with the code in this repo and then realized that the image of the traffic-manager and traffic-agent were `ambassador-telepresence-manager` and `ambassador-telepresence-agent`.

So I guess the bug is in the `ambassador-telepresence-agent` code not the open-source version.

BTW I find the doc unclear about these open-source vs. closed source images. The README for the helm chart (https://github.com/telepresenceio/telepresence/tree/release/v2/charts/telepresence) says the default value for `image.name` is `tel2` but that's not the case.
